ui/src/main/java/edu/wpi/grip/ui/pipeline/StepController.java

```diff
@@ -82,9 +100,21 @@ private void initialize() {
         new Image(InputStream.class.cast(icon))));
     buttons.getChildren().add(0, exceptionWitnessResponderButtonFactory.create(step, "Step Error"));
 
+    if (step.getInputSockets().stream()
+        .allMatch(inputSocket -> inputSocket.getSocketHint().getView()
+            .equals(SocketHint.View.NONE))) {
+      expand.setManaged(false);
+    } else {
+      expandIcon.setImage(new Image("/edu/wpi/grip/ui/icons/up.png"));
```

Comment:
Can you cache the image in the object so it isn't loaded from disk every time you click the button?